File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        # find the maximum pile
        maxPile = float('-inf')
        for pile in piles:
            if pile > maxPile:
                maxPile = pile
        # set left and right pointers
        left = 1
        right = maxPile
        speed = float('inf')

        while left <= right:
            time = 0
            mid = (left + right) // 2
            for pile in piles:
                time += math.ceil(pile / mid)

            if time > h:
                left = mid + 1
            
            if time <= h:
                speed = min(mid, speed)
                right = mid - 1
        
        return speed
            

            


    
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int len = numbers.length;
        int left = 0;
        int right = len - 1;
        while (left < right) {
            int curr = numbers[left] + numbers[right];
            if (curr > target) {
                right--;
            } else if (curr < target) {
                left++;
            } else {
                return new int[]{left + 1, right + 1};
            }
        }
        return new int[]{0,0};
    }
}
